### ISSUE TITLE
[BUG] Upgrade can be started while previous upgrade still in progress

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#6144]
**Describe the bug**
While upgrading from 1.2.1 to 1.2.2 after the first node was upgraded, there is already an upgrade button displayed in the UI, which allows starting the upgrade to 1.3.

**Expected behavior**
It should not be possible to start a new upgrade while the upgrade is still in progress.

<img width="1158" alt="Bildschirmfoto 2024-07-09 um 11 03 02" src="https://github.com/harvester/harvester/assets/3813921/f76efba4-ae05-4f9f-913f-408033ceea52">

<img width="1161" alt="Bildschirmfoto 2024-07-09 um 11 05 34" src="https://github.com/harvester/harvester/assets/3813921/e463f3a2-f5e4-4ca9-a4aa-ebe848736c0d">

**Comment by FrankYang0529** [comment]
@a110605 @torchiaf, may you take a look for this issue? In backend, we deny a new Upgrade request if any upgrade is in progress. We can also add this check to UI:

If there is any upgrade's label `harvesterhci.io/upgradeState` is not `Succeeded` and `Failed`, UI doesn't show new `Upgrade` button.

https://github.com/harvester/harvester/blob/8c709bac3c6e0ae54268298e49851eadf7af281a/pkg/webhook/resources/upgrade/validator.go#L122-L134

**Comment by torchiaf** [comment]
Thanks @FrankYang0529 , it should be a quick fix, I'll take a look.

**Comment by w13915984028** [comment]
Validation test: **PASS**

(1) v132->v140-0918 version upgrade

(1.1) 2 versions objects are created
```
$ kubectl get version.harvesterhci.io -A
NAMESPACE          NAME                    ISO-URL                                                            RELEASEDATE   MINUPGRADABLEVERSION
harvester-system   h-v14-master-20240925   http://192.168.2.59:8000/h-v14-master-20240925.iso                 20240925      
harvester-system   v1.4.0-dev-20240918     http://192.168.2.59:8000/harvester-v1.4.0-dev-20240918-amd64.iso   20240918      
```

(1.2) In the upgrade beginning stage, no addtional `Upgrade` button is shown
![image](https://github.com/user-attachments/assets/874fd3a2-1183-4eb9-9ba6-09a403561aa3)

(1.3) Manual trigger upgrade action is blocked by webhook:
```
cat > upgrade.yaml << EOF
apiVersion: harvesterhci.io/v1beta1
kind: Upgrade
metadata:
  generateName: hvst-upgrade-
  namespace: harvester-system
spec:
  image: ""
  logEnabled: false
  version: h-v14-master-20240925
EOF

$ kubectl create -f upgrade.yaml 
Error from server (Conflict): error when creating "upgrade.yaml": admission webhook "validator.harvesterhci.io" denied the request: cannot proceed until previous upgrade "hvst-upgrade-2bzml" completes
```

(1.4) After the node is rebooted, the upgrade menue is visiable.
![image](https://github.com/user-attachments/assets/b1fc93f4-1ebc-4584-9b51-98f3ebb3da96)


BTW: Another issue is observed and will be recorded on a new ticket.
